app/models/course.rb

```diff
@@ -4,8 +4,27 @@ class Course < ApplicationRecord
   belongs_to :user
   has_many :lessons
 
-  validates :description, :level, presence: true,
-                                  numericality: { only_integer: true, greater_than_or_equal_to: 1, less_than: 6 }
+  attr_accessor :video_link
+
+  before_save :take_video_id
+
+  validate :check_url
+  validates :description, :level, presence: true
+  validates :level, presence: true, numericality: { only_integer: true, greater_than_or_equal_to: 1, less_than: 6 }
   validates :name, presence: true, uniqueness: true
   validates :name, uniqueness: { case_sensitive: false }
+
+  private
+
+  def take_video_id
+    video_link.nil? ? nil : self[:youtube_video_id] = video_link.split('=').last
```

Comment:
ни где не видел такого подхода, это можно сделать проще, тк у модели есть уже есть setter для этого столбца
```suggestion
    video_link.nil? ? nil : self.youtube_video_id = video_link.split('=').last
```

да и вообще, тут тернарный оператор не нужен, лучше сделать так
```suggestion
    self.youtube_video_id = video_link.split('=').last if video_link
```

Reply:
я хочу оставить так для понимания

Reply:
"я пишу так потому что мне так понятнее" - какой смысл мне тогда проводить ревью?

Reply:
исправил